### DISCUSSION TITLE
VSCode: Jump to definition in types

### DISCUSSION THREAD

**Discussion by Xetoxyc** [prisma/prisma#22378] [Q&A]
### Question

I have a strange behaviour jumping into Prisma type structures. I usually use WebStorm and i do not have the problem there. Maybe its more a VSCode related issue.

In this sample, obviously its missing something, i can use the "Jump to definition (F12)" on `data`
```js
const jumpToWork = await this.prisma.person.create({
      data: {
        
      },
    });
```

In this sample, correct working one, the "Jump to definition (F12)" on `data` stops working. Means i cannot check anymore which fields are available.

```js
const jumpToWork = await this.prisma.person.create({
      data: {
        firstname: "a",
        tenantId: "1",
      },
    });
```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
enum PersonType {
  NATURAL_PERSON
  LEGAL_ENTITY
}

model Person {
  id   String     @id @default(uuid())
  firstname   String?
  lastname    String?

  tenantId String
  
  @@map("persons")
}
```

**Comment by ludralph** [comment]
Hi @Xetoxyc 👋 

Thank you for raising this question.

On VScode, accessing the properties in the `data` section, you can use `ctrl spacebar` and it will bring up a list of the properties for `data`.

I also tried using jump to definition for `data` property in Vscode, but it didn't work but I'll want to assume that is the default behaviour.
 
Meanwhile, you may want to share a recording of this behavior on either IDEs so I can share it internally with the team.

**Reply by Xetoxyc** [reply]
Hi @ludralph

Thank you for your response. I saw, that the `ctrl+spacebar` feature is working and it also show the type when hovering. Sometimes is just a bit complicated with deeper nested things. 

I have just made a quick video to demonstrate the WebStorm functionality. Basically its the same for every level of the type

https://github.com/prisma/prisma/assets/7449973/3f0826ac-acc4-44d3-9092-3f3a4173ec59

https://github.com/prisma/prisma/assets/7449973/3142c884-8994-4183-903a-03271816d75a

**Reply by Xetoxyc** [reply]
Hi @ludralph 
i forgot to mention, it only does not work if data is in a "correct" state. If you have a empty object as data for example it works totally fine as the correct type seems to be linked instead of an "inline" one

https://github.com/prisma/prisma/assets/7449973/ebfb671d-65c4-4d72-87b2-a898276ac887

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
